x-pack/plugins/dataset_quality/public/components/dataset_quality/dataset_quality.tsx

```diff
@@ -34,24 +40,33 @@ export const createDatasetQuality = ({ core, plugins }: CreateDatasetQualityArgs
     );
 
     return (
-      <DatasetQualityContext.Provider value={datasetQualityProviderValue}>
-        <KibanaContextProviderForPlugin>
-          <DatasetQuality />
-        </KibanaContextProviderForPlugin>
-      </DatasetQualityContext.Provider>
+      <SummaryPanelProvider
```

Comment:
Instead of having a new Context for SummaryPannel I think we should use [Actors](https://stately.ai/docs/actors) and invoke the summaryPannel machine from our principal one.